addons/mass_mailing_event/models/event_registration.py

```diff
@@ -9,4 +11,8 @@ class EventRegistration(models.Model):
     _mailing_enabled = True
 
     def _mailing_get_default_domain(self, mailing):
-        return [('state', 'not in', ['cancel', 'draft'])]
+        default_domain = [('state', 'not in', ['cancel', 'draft'])]
+        default_mailing_model_id = self.env.context.get('mailing_model_id')
+        if mailing.mailing_model_id.id == default_mailing_model_id and self.env.context.get('mailing_domain'):
```

Comment:
Just to avoid issues when everything is False / None
```suggestion
        if default_mailing_model_id and mailing.mailing_model_id.id == default_mailing_model_id and self.env.context.get('mailing_domain'):
```